docs/helm-chart/index.rst

```diff
@@ -140,3 +140,22 @@ will not start as the migrations will not be run:
 This is so these CI/CD services can perform updates without issues and preserve the immutability of Kubernetes Job manifests.
 
 This also applies if you install the chart using ``--wait`` in your ``helm install`` command.
+
+.. note::
+    While deploying this Helm chart with Argo, you might encounter issues with database migrations not running automatically on upgrade.
+
+To ensure database migrations with Argo CD, you will need to add:
+
+.. code-block::yaml
+
+    migrateDatabaseJob:
+        jobAnnotations:
+            "argocd.argoproj.io/hook": Sync
+
+and, equivalent configurations in case of Flux CD
+
+.. note::
+    This will ensure database migrations run when the Airflow Docker image is upgraded. This approach has a limitation in that the database migrations will run every time there is a ``Sync`` event in Argo. This is a trade-off for automation at the cost of some computational loss.
+
+.. warning::
+    While deploying with Argo CD, you may encounter issues with (built-in) Redis secrets (when you use the Celery(Kubernetes)Executor) not being rotated properly after Airflow pods restart. It's recommended that you bring your own Redis. See `Celery Backend <production-guide#celery-backend>`__
```

Comment:
And for the Argo CD case, instead of jumping straight to suggesting BYO redis, it might be better to suggest folks set `redis.passwordSecretName` and `data.brokerUrlSecretName`, or `redis.password` as a first option?

Reply:
I will make this change